src/panels/lovelace/cards/types.ts

```diff
@@ -139,6 +139,8 @@ export interface MapCardConfig extends LovelaceCardConfig {
   aspect_ratio?: string;
   default_zoom?: number;
   entities?: Array<EntityConfig | string>;
+  history_entities?: Array<EntityConfig | string>;
```

Comment:
Is adding a new array type better or would it make more sense to add it the entity config in the entities array
```yaml
type: map
entities:
   - entity: location.entity
      show_history: true
```

Reply:
Wow, thanks for the fast review!
I tried to seperate this, to allow geocoded_location entities without a device_tracker entity and vice versa. 
To retrieve the required information the cache filters for _geocoded_location entities. So it is maybe confusing to provide another entity in the configuration than the entity is used to retrieve the information.

Reply:
Why do you use the `geocoded_location` sensors and not the `device_tracker` history?

Reply:
Bram is right, we should just query the history of the actual entity that we're showing.

Reply:
I tried to do so. That was my first try and is its the best solution. 
But the attributes of an entity are only saved on state changes (as far as I know), and the state of a device_tracker ist unfortunatley bounded to zones. So the resulting path would only connect zones. 

>
> - The source entity has to be a _geocoded_location, because its state changes independent from any zones. To use the device_tracker as data source, a mobile device needs to provide a sort of location entity with a location (latitude, longitude and maybe altitude) as state, so we can access its history. This requires changes in the backend and the mobile apps as far as I can see, and can be a future enhancement to this feature

Reply:
The device tracker is not bound to zones. It depends on how you have configured the app to report data. I'm surprised that the geocoded sensor is more reliable. 

We should not allow for different history entities. That' working around a problem with the data in our system. Instead, we should just make sure that our system records history correctly. Otherwise we're going to push this problem to every user that has to fix it, instead of us being able to just make it work.

I think that we should limit this feature to just 1 config option: `hours_to_show`. If not set, show no history. If set, show that time of history.

Reply:
> 
> I think that we should limit this feature to just 1 config option: `hours_to_show`. If not set, show no history. If set, show that time of history.

I talked to @Santobert and he provides the backed change to realize your suggestion